### PR TITLE
Add pybindings for Spherepack

### PR DESCRIPTION
**Pull request by nikwit**
## Proposed changes

Adds some pybindings for `Spherepack`, mostly `phys_to_spec` and `spec_to_phys`. I also added bindings for the `SpherpackIterator` to write a sensible unit test comparing to scipy.

### CODE REVIEW

File: src/NumericalAlgorithms/SphericalHarmonics/Python/Spherepack.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                           const>(&ylm::Spherepack::phys_to_spec),
+           py::arg("collocation_values"), py::arg("physical_stride") = 1,
+           py::arg("physical_offset") = 0)
+      .def("spec_to_phys",
+           static_cast<DataVector (ylm::Spherepack::*)(
+               const DataVector& spectral_coefs, const size_t spectral_stride,
+               const size_t spectral_offset) const>(
+               &ylm::Spherepack::spec_to_phys),
+           py::arg("collocation_values"), py::arg("physical_stride") = 1,
+           py::arg("physical_offset") = 0)
```
Comment (nilsvu) [inline src/NumericalAlgorithms/SphericalHarmonics/Python/Spherepack.cpp]: fix argument names

File: tests/Unit/NumericalAlgorithms/SphericalHarmonics/Python/Test_Spherepack.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import numpy.testing as npt
+from scipy.special import sph_harm
+
+from spectre.SphericalHarmonics import Spherepack, SpherepackIterator
+
+
+def convert_coef_to_spherepack(complex_coef, l, m):
+    sign = 1.0 if m % 2 == 0 else -1.0
+    part = np.real(complex_coef) if m >= 0 else np.imag(complex_coef)
+    return 0.5 * sign * np.sqrt(2.0 / np.pi) * part
```
Comment (nilsvu) [inline tests/Unit/NumericalAlgorithms/SphericalHarmonics/Python/Test_Spherepack.py]: Is this conversion documented somewhere? If so, could you link to those docs here?

Edit: Looks like this is documented at https://spectre-code.org/classylm_1_1Spherepack.html, except I don't see the factor of 0.5.

**Review by nilsvu — CHANGES_REQUESTED** [review]
nice!

File: tests/Unit/NumericalAlgorithms/SphericalHarmonics/Python/Test_Spherepack.py (hunk 2 above)
Comment (nikwit) [reply tests/Unit/NumericalAlgorithms/SphericalHarmonics/Python/Test_Spherepack.py]: yes, I suspect I am counting double by just taking the real part of the complex harmonic. I will check that and update.

Comment (nikwit) [reply tests/Unit/NumericalAlgorithms/SphericalHarmonics/Python/Test_Spherepack.py]: I was indeed and corrected it in a fixup

**Review by nilsvu — APPROVED** [review]

**Pull request merged**